core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -645,6 +647,7 @@ public void execute(Runnable command) {
                 })
             .build();
     this.nameResolver = getNameResolver(target, nameResolverFactory, nameResolverArgs);
+    this.isAuthorityOverridden = builder.getOverrideAuthority() != null;
```

Comment:
This is splitting the obligation of managing the authority override. Let's move the `OverrideAuthorityNameResolverFactory` wrapping to ManagedChannelImpl (out of the builder). We should remove the `getNameResolverFactory()` method at the same time and just reference `builder.nameResolverFactory` directly.  (We could also drop OverrideAuthorityNameResolverFactory entirely and just have an if when determining the authority to pass to RealChannel, author using the override seems safer so we don't accidentally use the wrong authority.)